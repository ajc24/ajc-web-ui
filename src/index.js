/* Export all of the header components */
export { default as Header } from './ui-elements/header/Header';

/* Export all of the icon components */
export { default as SmallDownArrowIcon } from './ui-elements/icons/SmallDownArrowIcon';
export { default as SmallRightArrowIcon } from './ui-elements/icons/SmallRightArrowIcon';
