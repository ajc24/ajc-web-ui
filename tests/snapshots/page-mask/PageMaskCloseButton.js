import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { PageMaskCloseButton } from '../../../src';

describe('PageMaskCloseButton', () => {
  describe('Default props and rendering - onClose functionality not set', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <PageMaskCloseButton id="test-id" />
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Default component with onClose functionality set', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <PageMaskCloseButton id="test-id" onClose={() => {}} />
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Red button with onClose functionality set', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <PageMaskCloseButton id="test-id" colour="red" onClose={() => {}} />
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
