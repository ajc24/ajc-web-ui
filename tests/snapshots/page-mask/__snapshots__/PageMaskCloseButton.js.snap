// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageMaskCloseButton Default props and rendering - onClose functionality not set verifies the snapshot for the component 1`] = `null`;

exports[`PageMaskCloseButton Transferred props and rendering - Default component with onClose functionality set verifies the snapshot for the component 1`] = `
<button
  className="ajc-flex-row ajc-button ajc-button-page-mask-close ajc-text-white ajc-font-default ajc-button-grey"
  id="test-id"
  onClick={[Function]}
  title="Click to close"
>
   X 
</button>
`;

exports[`PageMaskCloseButton Transferred props and rendering - Red button with onClose functionality set verifies the snapshot for the component 1`] = `
<button
  className="ajc-flex-row ajc-button ajc-button-page-mask-close ajc-text-white ajc-font-default ajc-button-red"
  id="test-id"
  onClick={[Function]}
  title="Click to close"
>
   X 
</button>
`;
